proposals/0106-paritioned-epoch-rewards-pda.md

```diff
@@ -0,0 +1,100 @@
+---
+simd: '0106'
+title: Epoch Rewards Partition Data PDA
+authors:
+  - Tyera Eulberg
+category: Standard
+type: Core
+status: Draft
+created: 2024-01-17
+feature: https://github.com/solana-labs/solana/issues/32166
+extends: 0015
+---
+
+## Summary
+
+Partitioned epoch rewards (SIMD 0015) will split epoch rewards distribution
+across multiple blocks at the start of an epoch. This extension of that SIMD
+proposes storing data about the rewards calculation and partitions in a sysvar
+account at a program-derived address (PDA), which can be queried by clients, or
+indeed within the runtime, to performantly identify the partition for the
+distribution to any particular address.
+
+## Motivation
```

Comment:
Can you also include the motivation for including this data on-chain? I personally think that storing it in the first blocks metadata would be great. We already use the metadata to record the list of rewards.

Reply:
Added. But yes, planning to do both. Will write up a separate SIMD for the metadata approach.

Reply:
I reverted this one as well. Here's what I had written:
"If the partition data were stored in an on-chain account, it would be simple and
low-compute-cost for a client (with a snapshot or running node) or the runtime
itself to find the rewards for a particular address for any historical epoch
after partitioned epoch rewards are activated."

I do feel like this is already articulated in the summary.

Reply:
> Ok, how open are you to removing the PDA approach and just using the metadata?

Somewhat opposed. I think it makes sense to have some sort of persistent record of the partitioned-rewards data on chain, but it's true I don't know of a specific in-program use-case for it. From the rpc perspective, it's also really nice to be able to figure out the rewards block before having to determine what data store to use to find that block.
@t-nelson , perhaps you'd care to chime in here?

Reply:
Is that very different from the rpc fetching the first block in the epoch to find out where the rewards are? If I recall correctly, there's an extra call getBlocks to figure out what the first block actually is, is that the part you're trying to improve?

Reply:
> there's an extra call getBlocks to figure out what the first block actually is, is that the part you're trying to improve?

Yes. There's potentially quite a bit of latency in each long-term storage query, so it would be nice to avoid the extra call.

Reply:
If the main motivation is to save some latency overhead on step 1 of fetching rewards when step 2 requires high latency big table fetching most of the time anyways, it doesn't seem like this proposal is worth the runtime complexity.